pkg/pipeline/encode/encode_prom.go

```diff
@@ -28,49 +29,56 @@ import (
 	"net/http"
 	"os"
 	"strconv"
+	"sync"
+	"time"
 )
 
-// TODO when do we clean up old entries?
+const defaultExpiryTime = 120
 
-type gaugeInfo struct {
-	input     string
-	tags      []string
-	promGauge *prometheus.GaugeVec
-}
-
-type counterInfo struct {
-	input     string
-	tags      []string
+type genericMetricInfo struct {
```

Comment:
you can just change the name the metricInfo -- the `generic` is not really needed

Reply:
Done